modules/data_release/jsx/uploadFileForm.js

```diff
@@ -0,0 +1,218 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import ProgressBar from 'ProgressBar';
+
+
+/**
+ * Upload File Form
+ *
+ * Module component rendering the upload file form modal window
+ */
+class UploadFileForm extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      data: {},
+      formData: {},
+      uploadResult: null,
+      errorMessage: {},
+      hasError: {},
+      isLoaded: false,
+      uploadProgress: -1,
+    };
+
+    this.updateFormElement = this.updateFormElement.bind(this);
+    this.validateAndSubmit = this.validateAndSubmit.bind(this);
+    this.uploadFile = this.uploadFile.bind(this);
+  }
+
+  componentDidMount() {
+    let self = this;
+    $.ajax(this.props.DataURL, {
```

Comment:
fetch

Reply:
Sure! Is that enough balls? LOL

![image](https://user-images.githubusercontent.com/1402456/67695984-69ee5a00-f97c-11e9-808b-276e8c1ed1e4.png)